### PR TITLE
Ch63778/implement cancel order ajax callback

### PR DESCRIPTION
**Pull request by N-Molham**
# Summary

This PR implements and register `AJAX::admin_cancel_order()` as AJAX callback for `AJAX::ACTION_CANCEL_ORDER` action as detailed [here](https://docs.google.com/document/d/11AG_ZVIYGKn96BvEn_Qwy91FOALeE7GmB5cpfN1D6LU/edit#heading=h.1dj1smskr86c)

## UI Changes

None

## QA

TODO write `AJAX_Test::test_admin_cancel_order` Integration test as I am unable to test it without API calls

## Before merge

- [x] I have confirmed these changes in each supported minor WooCommerce version

### CODE REVIEW

File: includes/AJAX.php [hunk 1]
```diff
@@ -49,6 +51,58 @@ public function __construct() {
...
+
+			wp_send_json_success( 'cancelled' );
+
+		} catch ( Framework\SV_WC_Plugin_Exception $exception ) {
+
+			if ( $order instanceof \WC_Abstract_Order ) {
+				$order->add_order_note( sprintf( __( 'Could not cancel order. Code: "%s", message: "%s"', 'facebook-for-woocommerce' ), $exception->getCode(), $exception->getMessage() ) );
+			}
+
+			wp_send_json_error( $exception->getMessage() );
```
Comment (unfulvio) [inline includes/AJAX.php:104]: the exceptions don't seem having a code, though, and the message is not translatable - if  exceptions messages are meant to be displayed to a user other than a developer they could probably be translated via `__()`

also the `sprintf()` here is missing the `/* translators: Placeholders: */` comment to help translators with the placholders content

Comment (N-Molham) [reply includes/AJAX.php:104]: @unfulvio understood ... the implementation doc suggested to include the error code as well ... but I agree it doesn't look like any of the errors have codes

Comment (N-Molham) [reply includes/AJAX.php]: @unfulvio see new commits below, They should address your comments

**Review by unfulvio — APPROVED** [review]
looks better, I made a couple of edits (check commits below)

**Pull request merged**